app/src/main/java/hpsaturn/pollutionreporter/BaseActivity.java

```diff
@@ -24,20 +27,39 @@
 public abstract class BaseActivity extends RxAppCompatActivity {
 
     public static String TAG = BaseActivity.class.getSimpleName();
+    
+    private void enableBLE() {
+        Logger.i(TAG,"[BLE] enableBLE: starting...");
+
+        ActivityResultLauncher<Intent> someActivityResultLauncher = registerForActivityResult(
+                new ActivityResultContracts.StartActivityForResult(),
+                result -> {
+
+                    if (result.getResultCode() == Activity.RESULT_OK) {
+                        //Intent data = result.getData();
+                        Logger.i(TAG,"[BLE] enableBLE: enabled!");
+                    } else {
+                        Logger.i(TAG,"[BLE] enableBLE: not enabled!");
+                    }
+                });
+        
+        Intent enableBtIntent = new Intent(BluetoothAdapter.ACTION_REQUEST_ENABLE);
+        someActivityResultLauncher.launch(enableBtIntent);
+    }
 
-    public void checkBluetoohtBle() {
+    public void checkBLE() {
         // Use this check to determine whether BLE is supported on the device.
         BluetoothManager bluetoothManager = (BluetoothManager) getSystemService(Context.BLUETOOTH_SERVICE);
         BluetoothAdapter mBluetoothAdapter = bluetoothManager.getAdapter();
         if (!getPackageManager().hasSystemFeature(PackageManager.FEATURE_BLUETOOTH_LE)) {
-            Toast.makeText(this, R.string.msg_ble_not_supported, Toast.LENGTH_SHORT).show();
+            Toast.makeText(this, this.getResources().getString(R.string.msg_ble_not_supported), Toast.LENGTH_SHORT).show();
```

Comment:
I don't understand this change. makeText support pass the string identifier or int. The old implementation use it, and was more short

Reply:
It think it's ok without "this.getResources().getString", you can revert this change.